### PR TITLE
Check that functions exist in isoltest.

### PR DESCRIPTION
**Pull request by chriseth**
References https://github.com/ethereum/solidity/pull/8294

### CODE REVIEW

**Comment by chriseth** [comment]
Tests fail because the deliberately call non-existing functions.

We can either (1) reserve a specific name for non-existing functions, (2) move those tests out of isoltest or (3) test this behaviour from within a solidity contract.

**Comment by ekpyron** [comment]
Or (4), we can abuse our fallback mechanism ``(): hex"12345678" -> FAILURE`` or (5) we could have a test setting for disabling the validation.

File: test/libsolidity/SemanticTest.cpp [hunk 1]
```diff
@@ -143,13 +143,22 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
...
 			{
-				bytes output = test.call().useCallWithoutSignature ?
-					callLowLevel(test.call().arguments.rawBytes(), test.call().value) :
-					callContractFunctionWithValueNoEncoding(
+				bytes output;
+				if (test.call().useCallWithoutSignature)
+					output = callLowLevel(test.call().arguments.rawBytes(), test.call().value);
+				else
+				{
+					if (!m_compiler.methodIdentifiers(m_compiler.lastContractName()).isMember(test.call().signature))
```
Comment (ekpyron) [inline test/libsolidity/SemanticTest.cpp]:
```suggestion
					soltestAssert(m_compiler.methodIdentifiers(m_compiler.lastContractName()).isMember(test.call().signature), "The function " + call.call().signature + " is not known to the compiler");
```
?

The exception should be caught and reported... it's not so different from the constructor assertions above, is it?

Comment (chriseth) [reply test/libsolidity/SemanticTest.cpp]: Failure to deploy a contract cannot be attributed to a single test, while this one can.

**Review by ekpyron — APPROVED** [review]

File: test/libsolidity/SemanticTest.cpp (hunk 1 above)
Comment (ekpyron) [reply test/libsolidity/SemanticTest.cpp]: On the other hand an invalid constructor call *can* be attributed to a single expectation line...

I'd say everything upon which you can't just hit "update" is a ``soltestAssert``ion failure - and you can't update expectations for invalid signatures.

**Pull request merged**